### DISCUSSION TITLE
Can this be used on Books?

### DISCUSSION THREAD

**Discussion by wisnoskij** [pndurette/gTTS#368] [Q&A] [answered]
Hello, Fantastic project.

Is this appropriate for use on very long text files such as books, or will I get my account or this tool in trouble?
Would I separate such files into chapters, pages, or paragraphs, or just send the entire book in one go?

**Comment by pndurette** [comment, marked as answer]
> Hello, Fantastic project.
> 
> 
> Is this appropriate for use on very long text files such as books, or will I get my account or this tool in trouble?
> 
> Would I separate such files into chapters, pages, or paragraphs, or just send the entire book in one go?

Thanks! Also thank you for the concern! So, it _would_ work but there is some rate limiting applied (read: being blocked for some time). So I would suggest indeed programmatically cutting in parts with some artificial pauses between them.

**Comment by wisnoskij** [comment]
Update for anyone interested.
It seems to work fine, but is fairly slow and their are still a few bugs to work out.
It took 14h to covert 1.5mb text to 1d of audio.
8h of this was waiting between files (5 minute wait time). I imagine this could be a lot lower.
This was separated into 100 files.

The system does break with `Error: 200 (OK) from TTS API. Probable cause: No audio stream in response. Unsupported language 'en'` occasionally. 
It looks like some character `&#x00c2;&#x00a0;` was causing this in some places, but it looks like other things are also causing the breakage in a few files. It looks like it breaks on the line with the bad data so it has been ok to debug, but I have no idea why it is breaking for one file. hopefully I just have to filter out all non standard characters.

Been working on a batch convert utility that I will upload after I get these bugs worked out.
